fairlearn/metrics/_metrics_engine.py

```diff
@@ -57,6 +58,7 @@ def metric_by_group(metric_function,
         result.overall = metric_function(y_a, y_p, **kwargs)
 
     groups = np.unique(group_membership)
+    result.by_group = dict()
```

Comment:
Bunch _is_ a `dict`, and it seems the only thing we're putting in it here is another `dict` inside `by_group`. Doesn't it then make sense to put those results directly into result instead of under by_group?

Reply:
That would be a problem if someone decides to have an 'overall' entry in their `group_membership` vector. Not sure why someone would, but that's either something else to check, or a charming pitfall for some user.

Reply:
But we can check for that, can't we?

Reply:
Yes - as I said it's something else to check. I just think that it would be a bit odd to restrict the keys a user could use in their data. Especially as they most likely won't notice that bit of the documentation, but will instead end up with a sudden exception.

Reply:
I had a same thought as @adrinjalali . I think we should try to make the result shallow. Let's say that the two groups are "male" and "female". Then we could use keys like:
"group_male", "group_female", "overall", "min", "max", "range", "range_ratio"

Reply:
I've updated this again. Please take a look at `test_metrics_engine.py` and let me know if you like the new syntax. I would like to have this agreed before going through and changing all of the other tests.

@MiroDudik @adrinjalali

Reply:
I was discussing this with @rihorn2  at lunch, and I like this even less. Going to this level of change will require changes to the dashboard too. I do not see the benefits.

Reply:
it'd be nice if people who are not there in your office could see the result of your discussions in a better way :)

Reply:
The result of the discussion: what is the benefit of doing this ('this' being _any_ of this change)?

Reply:
the hope was to better support the use cases outlined in: https://github.com/fairlearn/fairlearn/issues/257#issuecomment-580811906
and also remove frictions that come about when you introduce new data types:
https://github.com/fairlearn/fairlearn/issues/257#issuecomment-581414697